backend/src/campaigns.rs

```diff
@@ -157,3 +158,42 @@ pub async fn roles(
 
     Ok(Json(RolesResponse { roles }))
 }
+
+#[post("/<campaign_id>/roles", data = "<role>")]
```

Comment:
I hate to do this, but I'm thinking the route might actually make more sense under post "/role/new", data="<role>" in role.rs

Then, it'd mean RoleInput and RoleUpdate can be the same object and it'd be easier to find the route.

If you could change that route here, and in the confluence doc give it a quick edit), I'd really appreciate it.
(it would also avoid importing things like RolesError to campaigns.rs which would probably help tbh)